docker/docker_image_src.go

```diff
@@ -83,6 +83,26 @@ func (s *dockerImageSource) GetManifest() ([]byte, string, error) {
 	return s.cachedManifest, s.cachedManifestMIMEType, nil
 }
 
+// GetTargetManifest ...
+func (s *dockerImageSource) GetTargetManifest(reference string) ([]byte, string, error) {
```

Comment:
There really should be a single function with this code, shared by `GetTargetManifest` and `ensureManifestIsLoaded`.